### PR TITLE
Fixes for Gemini inside Vertex

### PR DESCRIPTION
**Pull request by tbeamish-benchsci**
# Description

1) Gemini requires an even message count and, in the Gemini LLM, we `merge_neighboring_same_role_messages`. This PR does the same thing in the Vertex LLM if we're using Gemini.

2) We also set the default `system_role` in the  `LLMMetadata` to be `System.USER` only if we're using Gemini, since Gemini only supports "user" and "model" roles.

3) Finally we ensure the role of each message in Vertex is set to the appropriate `MessageRole`

**NOTE: This change adds/updates NO UNIT TESTs to support it.**

Fixes # (https://github.com/run-llama/llama_index/issues/11439)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from llama_index.core.llms.callbacks import llm_chat_callback, llm_completion_callback
 from llama_index.core.llms.llm import LLM
 from llama_index.core.types import BaseOutputParser, PydanticProgramMode
+from llama_index.llms.gemini.utils import merge_neighboring_same_role_messages
```
Comment (tbeamish-benchsci) [inline llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py:19]: This places a dependency on Gemini from Vertex - which prolly isn't ideal.
An alternative would be to copy this function over to `llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/gemini_utils.py`

But an even better solution would be to move it to some common folder that both Gemini and Vertex can import from. Where is that folder?

File: llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py [hunk 2]
```diff
@@ -105,7 +106,7 @@ def merge_neighboring_same_role_messages(
 
         # Create a new ChatMessage or similar object with merged content
         merged_message = ChatMessage(
-            role=current_message.role,
+            role=ROLES_TO_GEMINI[current_message.role],
```
Comment (tbeamish-benchsci) [inline llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py:109]: This fixes Vertex but unsure if it'll break Gemini. Do we have a unit test that will tell us?

File: llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py:19]: probably this function could be moved to core, in some LLM utils file?

File: llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py (hunk 2 above)
Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py:109]: nope, you would have to actually call the API to confirm it works (which tbh its probably fine?)

Comment (tbeamish-benchsci) [reply llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py:109]: I don't have access to Gemini directly, only via Vertex.

**Review by logan-markewich — APPROVED** [review]
this looks good to me, thanks @tbeamish-benchsci !

**Pull request merged**